yoyodyne/sizing.py

```diff
@@ -0,0 +1,96 @@
+"""Finds the best batch size given constraints."""
+
+import argparse
+from typing import Tuple
+
+import pytorch_lightning as pl
+from pytorch_lightning.tuner import tuning
+
+from . import data, defaults, models
+
+
+def max_batch_size(
+    trainer: pl.Trainer,
+    model: models.BaseEncoderDecoder,
+    datamodule: data.DataModule,
+    *,
+    mode: str = defaults.AUTO_BATCH_SIZE_MODE,
+    steps_per_trial: int = defaults.AUTO_BATCH_SIZE_STEPS_PER_TRIAL,
+    max_trials: int = defaults.AUTO_BATCH_SIZE_MAX_TRIALS,
+) -> int:
+    """Computes the maximum batch size that will fit in memory."""
+    tuner = tuning.Tuner(trainer)
+    return tuner.scale_batch_size(
+        model,
+        datamodule=datamodule,
+        mode=mode,
+        steps_per_trial=steps_per_trial,
+        max_trials=max_trials,
+    )
+
+
+def optimal_batch_size(
+    desired_batch_size: int, max_batch_size: int
+) -> Tuple[int, int]:
+    r"""Computes optimal batch size and number of gradient accumulation steps.
+
+    Given the desired batch size $b$ and a max batch size $n_{\textrm{max}}$,
+    this returns a tuple $a \in \mathcal{N}_{+}, n \in \mathcal{N}_{\le n}$
```

Comment:
Should `n \in \mathcal{N}_{\le n}` --> `n \in \mathcal{N}_{\le n_\textrm{max}}`?

Reply:
+1